### PR TITLE
add tauid shifts for all eras

### PR DESCRIPTION
**Pull request by harrypuuter**
Adding up and down shifts for all eras for TauID:

et and mt: one Variation per pT bin ([30., 35., 40., 500., 1000. ,$\le$ 1000.])

tt: one Variation per DM ([0,1,10,11})

add TauES Shifts for DM 11

### CODE REVIEW

**Review by janekbechtel — APPROVED** [review]

File: shapes/produce_shapes_2016.py [hunk 1]
```diff
@@ -620,6 +620,64 @@ def readclasses(c):
...
+                tau_id_variations.append(
+                    AddWeight("CMS_eff_tauid_{}-{}_{}_Run2016".format(bindown, binup, channel), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel),
+                            Weight("(((pt_2 >= {bindown})*tauIDScaleFactorWeightDown_tight_DeepTau2017v2p1VSjet_2*1.0/tauIDScaleFactorWeight_tight_DeepTau2017v2p1VSjet_2)+((pt_2 < {bindown})*1.0))".format(bindown=bindown), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel)), "Down"))
+            else:
+                tau_id_variations.append(
+                AddWeight("CMS_eff_tauid_{}-{}_{}_Run2016".format(bindown, binup, channel), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel),
+                        Weight("(((pt_2 >= {bindown} && pt_2 < {binup})*tauIDScaleFactorWeightDown_tight_DeepTau2017v2p1VSjet_2*(1.0/tauIDScaleFactorWeight_tight_DeepTau2017v2p1VSjet_2))+((pt_2 < {bindown} && pt_2 > {binup})*1.0))".format(bindown=bindown, binup=binup), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel)), "Up"))
```
Comment (janekbechtel) [inline shapes/produce_shapes_2016.py]: Should be "(pt_2 < {bindown} || pt_2 > {binup})*1.0)"

Comment (harrypuuter) [reply shapes/produce_shapes_2016.py]: oh wow
you are so right - who told you about this bug ?

Comment (janekbechtel) [reply shapes/produce_shapes_2016.py]: ☝️

**Review by janekbechtel — APPROVED** [review]

File: shapes/produce_shapes_2016.py [hunk 2]
```diff
@@ -620,6 +620,64 @@ def readclasses(c):
...
+    for channel in ["et" , "mt"]:
+        pt = [30, 35, 40, 500, 1000, "inf"]
+        tau_id_variations = []
+        for i, ptbin in enumerate(pt[:-1]):
+            bindown = ptbin
+            binup = pt[i+1]
+            if binup == "inf":
+                tau_id_variations.append(
+                AddWeight("CMS_eff_tauid_{}-{}_{}_Run2016".format(bindown, binup, channel), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel),
+                        Weight("(((pt_2 >= {bindown})*tauIDScaleFactorWeightUp_tight_DeepTau2017v2p1VSjet_2*1.0/tauIDScaleFactorWeightUp_tight_DeepTau2017v2p1VSjet_2)+((pt_2 < {bindown})*1.0))".format(bindown=bindown), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel)), "Up"))
```
Comment (swozniewski) [inline shapes/produce_shapes_2016.py]: Isn't this just 1.0? `tauIDScaleFactorWeightUp_tight_DeepTau2017v2p1VSjet_2*1.0/tauIDScaleFactorWeightUp_tight_DeepTau2017v2p1VSjet_2`

Comment (harrypuuter) [reply shapes/produce_shapes_2016.py]: updated

File: shapes/produce_shapes_2016.py (hunk 1 above)
Comment (harrypuuter) [reply shapes/produce_shapes_2016.py]: updated

**Comment by harrypuuter** [comment]
I tested it for 2018 only, but 2017 and 2016 are a copy/paste

**Comment by harrypuuter** [comment]
updated for dm11, this PR is tied to this PR in the shape_producer: https://github.com/KIT-CMS/shape-producer/pull/27

**Review by swozniewski — APPROVED** [review]

**Pull request merged**